Marlin/src/libs/numtostr.cpp

```diff
@@ -384,6 +387,32 @@ const char* ftostr61rj(const_float_t f) {
   return &conv[1];
 }
 
+// Converts float to string representing its absolute value with up to 8 characters including decimal dot, right aligned, example: ____5678, ____5.67, ___45.67, __345.67, _2345.67, 12345.67
+/**
+ * IMPORTANT: Supports only floats that can be displayed by above mentioned format
+ * IMPORTANT: Float is represented by somewhat revered scientific notation using two parameters, from which float can be acquired by dividing: @param intVal / @param decimal
+ * @param intVal significant digits of float
+ * @param decimal divisor to acquire the float, must be power of 10
+ * */ 
+const char* ftostr7xrj(int32_t intVal, uint32_t decimal) {
+  
+  if (intVal < 0) intVal *= -1;
+
+  int8_t intEnd = 7;
+  uint8_t dig = 0;
+  uint32_t div = 1;
+  for (;intEnd >= 0 && (((float)intVal / div) >= 1 || ((float)decimal / div) >= 1);intEnd--) {
```

Comment:
Save a few cycles by skipping cast and division since…
- `((float)intVal / div) >= 1` => `(intVal >= div)`
- `((float)decimal / div) >= 1` => `(decimal >= div)`